testcontainers/src/core/image/image_ext.rs

```diff
@@ -48,6 +48,18 @@ pub trait ImageExt<I: Image> {
     /// Sets the network the container will be connected to.
     fn with_network(self, network: impl Into<String>) -> ContainerRequest<I>;
 
+    /// Adds the specified labels to the container.
+    ///
+    /// **Note**: in addition to all keys in the `com.testcontainers.*` namespace, there
+    /// are certain labels that are used by `testcontainers` internally which will always
+    /// be unconditionally overwritten, and so should not be expected or relied upon to
+    /// be applied correctly if they are included in `labels`. Currently, they are:
+    /// - `managed-by`
+    fn with_labels(
+        self,
+        labels: impl IntoIterator<Item = (impl Into<String>, impl Into<String>)>,
+    ) -> ContainerRequest<I>;
```

Comment:
Since it's append semantic, I'd suggest to add a singular form `with_label` (we may have both or just singular actually). Also singular methods seems aligned [with Java impl](https://java.testcontainers.org/features/advanced_options/#container-labels)

I mean a common use-case is to pass couple of labels in-place, but passing many of them also convinient with iterator so we can have both

WDYT?

Reply:
@DDtKey An excellent suggestion. Implemented and pushed 😁